src/index.js

```diff
@@ -1447,5 +1452,266 @@ module.exports = {
+      const isTextMatching = (value, valueSpec) => {
+        if (typeof valueSpec === 'string') {
+          if (value === valueSpec) {
+            return true;
+          }
+        } else if (valueSpec instanceof RegExp) {
+          if (valueSpec.test(value)) {
+            return true;
+          }
+        } else if (typeof valueSpec === 'function') {
+          return true;
+        }
+
+        return false;
+      };
+      const isHtml = isInsideHtmlDocument(element);
+
+      let score = 0;
+
+      const nodeName = isHtml
+        ? element.nodeName.toLowerCase()
+        : element.nodeName;
+
+      if (isTextMatching(nodeName, spec.name)) {
+        score++;
+      }
+
+      if (isTextMatching(element.textContent, spec.textContent)) {
+        score++;
+      }
+
+      if (typeof element.hasAttribute === 'function') {
+        const attributes = spec.attributes || {};
+        const className = attributes['class'];
+        const style = attributes.style;
+
+        if (className && element.hasAttribute('class')) {
+          if (typeof className === 'string') {
+            const expectedClasses = getClassNamesFromAttributeValue(className);
+            const actualClasses = getClassNamesFromAttributeValue(
+              element.getAttribute('class')
+            );
+
+            expectedClasses.forEach(expectedClass => {
+              if (actualClasses.indexOf(expectedClass) !== -1) {
+                score++;
+              }
+            });
+          } else if (isTextMatching(element.getAttribute('class'), className)) {
+            score++;
+          }
+        }
+
+        if (style && element.hasAttribute('style')) {
+          const expectedStyles =
+            typeof style === 'string' ? styleStringToObject(style) : style;
+          const actualStyles = styleStringToObject(
+            element.getAttribute('style')
+          );
+
+          Object.keys(expectedStyles).forEach(styleName => {
+            const expectedStyle = expectedStyles[styleName];
+            const actualStyle = actualStyles[styleName];
+
+            if (actualStyle) {
+              score++;
+            }
+
+            if (isTextMatching(actualStyle, expectedStyle)) {
+              score++;
+            }
+          });
+        }
+
+        const specialAttributes = ['style', 'class'];
+        const ids = ['id', 'data-test-id', 'data-testid'];
+
+        Object.keys(attributes).forEach(attributeName => {
+          if (specialAttributes.indexOf(attributeName) !== -1) {
+            return; // skip
+          }
+
+          if (element.hasAttribute(attributeName)) {
+            if (typeof attributes[attributeName] === 'boolean') {
+              score++;
+            }
+
+            if (
+              element.getAttribute(attributeName) === attributes[attributeName]
+            ) {
+              score += ids.indexOf(attributeName) !== -1 ? 100 : 1;
+            }
+          } else if (!attributes[attributeName]) {
```

Comment:
What about `spec.attributes.foo === ''`? Doesn't seem like that should be treated the same as if the attribute is absent from `spec.attributes`?